e2e-tests/smart-update/run

```diff
@@ -64,134 +68,152 @@ function check_last_pod_to_update {
+    for i in $(seq 0 $((CLUSTER_SIZE - 1))); do
+        compare_mysql_cmd "select-1" "SELECT * from myApp.myApp;" "-h ${CLUSTER}-pxc-${i}.${CLUSTER}-pxc -uroot -proot_password"
     done
 
-    kubectl_bin delete -f $test_dir/conf/$cluster-version-service-unreachable.yml
+    kubectl_bin delete -f "${tmp_dir}/${CLUSTER}-version-service-unreachable.yml"
     kubectl_bin delete pvc --all
 
+    ##################################################
     desc 'PXC cluster update with the latest image by version service'
-    spinup_pxc "$cluster" "$test_dir/conf/$cluster-version-service-unreachable.yml"
-    version_service_image="latest"
-    initial_primary=$(run_mysql 'SELECT @@hostname hostname;' "-h $cluster-haproxy -uroot -proot_password")
+    spinup_pxc "${CLUSTER}" "${tmp_dir}/${CLUSTER}-version-service-unreachable.yml"
+    vs_image="latest"
+    initial_primary=$(run_mysql 'SELECT @@hostname hostname;' "-h ${CLUSTER}-haproxy -uroot -proot_password")
 
-    kubectl_bin patch pxc/$cluster --type=merge -p '{"spec":{"upgradeOptions":{"versionServiceEndpoint":"https://check.percona.com/versions","apply":"'${version_service_image}'","schedule": "* * * * *"}}}'
-    target_image=$(get_vs_image ${version_service_image})
+    kubectl_bin patch pxc/"${CLUSTER}" --type=merge -p '{"spec":{"upgradeOptions":{"versionServiceEndpoint":"'${VS_ENDPOINT}'","apply":"'${vs_image}'","schedule": "* * * * *"}}}'
     sleep 55
 
-    check_last_pod_to_update ${cluster} ${initial_primary} ${pxc_size} ${target_image}
-
-    wait_cluster_consistency ${cluster} ${pxc_size} $(kubectl_bin get pxc ${cluster} -o jsonpath='{.spec.'$(get_proxy_engine ${cluster})'.size}')
-    for i in $(seq 0 $((${pxc_size} - 1))); do
-        compare_mysql_cmd "select-1" "SELECT * from myApp.myApp;" "-h ${cluster}-pxc-${i}.${cluster}-pxc -uroot -proot_password"
+    check_last_pod_to_update "${CLUSTER}" "${initial_primary}" "${CLUSTER_SIZE}" "${TARGET_IMAGE_PXC_VS}"
+    wait_cluster_consistency "${CLUSTER}" "${CLUSTER_SIZE}" "${PROXY_SIZE}"
+    for i in $(seq 0 $((CLUSTER_SIZE - 1))); do
+        compare_mysql_cmd "select-1" "SELECT * from myApp.myApp;" "-h ${CLUSTER}-pxc-${i}.${CLUSTER}-pxc -uroot -proot_password"
     done
 
-    kubectl_bin delete -f $test_dir/conf/$cluster-version-service-unreachable.yml
+    kubectl_bin delete -f "${tmp_dir}/${CLUSTER}-version-service-unreachable.yml"
     kubectl_bin delete pvc --all
 
+    ##################################################
     desc 'PXC cluster update with explicitly specified image inside version service'
-    spinup_pxc "$cluster" "$test_dir/conf/$cluster-version-service-unreachable.yml"
-    version_service_image="8.0.19-10.1"
-    initial_primary=$(run_mysql 'SELECT @@hostname hostname;' "-h $cluster-haproxy -uroot -proot_password")
+    spinup_pxc "${CLUSTER}" "${tmp_dir}/${CLUSTER}-version-service-unreachable.yml"
+    if [[ "${PXC_VER}" == "5.7" ]]; then
+        vs_image="5.7.31-31.45"
```

Comment:
I do not like the idea that we hardcode it. Can we avoid it?

Reply:
Thanks! Fixed.